src/Perl6/Actions.nqp

```diff
@@ -1675,27 +1675,27 @@ class Perl6::Actions is HLL::Actions does STDActions {
             $past := $<EXPR>.ast;
             if $mc {
                 if ~$mc<sym> eq 'with' {
-                    make thunkity_thunk($/,'.b',QAST::Op.new( :op('call'), :name('&infix:<andthen>')),[$mc,$<EXPR>]);
-                    return;
+                    $past := make thunkity_thunk($/,'.b',QAST::Op.new( :op('call'), :name('&infix:<andthen>')),[$mc,$<EXPR>]);
```

Comment:
Can these `make` calls be removed alongside the early `return`? There's another at the end of the method.

Reply:
Sure, I forgot these.